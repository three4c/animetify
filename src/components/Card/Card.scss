.Card {
  border-radius: 16px;
  background-color: #fff;
  box-shadow: 0 0 24px rgba(0, 0, 0, 0.02);
  padding: 24px 0;
  width: 100%;

  &:not(:last-child) {
    margin-bottom: 32px;
  }

  & &__time {
    display: inline-block;
    margin: 0 24px 8px;
  }

  & &__title {
    border-bottom: 1px solid #eee;
    padding-bottom: 16px;
    margin: 0 24px 16px;
  }
}
