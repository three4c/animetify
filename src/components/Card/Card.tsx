import React from 'react';

import './Card.scss';

interface CardProps {
  /** タイトル */
  title: string;
  /** 放送日 */
  broadcastStartDate: string;
}

const Card: React.FC<CardProps> = (props) => {
  console.log('a');

  return (
    <div className="Card util-focus" tabIndex={0}>
      <time className="Card__time" dateTime={props.broadcastStartDate}>
        {props.broadcastStartDate.replace(/-/g, '.')}
      </time>
      <h2 className="Card__title">{props.title}</h2>
      {props.children}
    </div>
  );
};
export default Card;
