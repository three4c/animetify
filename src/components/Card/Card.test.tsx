import React from 'react';

import { mount } from 'enzyme';

import Card from './Card';

describe('<Card />', () => {
  it('Card render', () => {
    expect(
      mount(
        <Card title="電波女と青春男" broadcastStartDate="2011-04-15">
          <div>Card Children</div>
        </Card>
      )
    ).toMatchSnapshot();
  });
});
