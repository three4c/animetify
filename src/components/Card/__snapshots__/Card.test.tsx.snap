// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Card render 1`] = `
<Card
  broadcastStartDate="2011-04-15"
  title="電波女と青春男"
>
  <div
    className="Card util-focus"
    tabIndex={0}
  >
    <time
      className="Card__time"
      dateTime="2011-04-15"
    >
      2011.04.15
    </time>
    <h2
      className="Card__title"
    >
      電波女と青春男
    </h2>
    <div>
      Card Children
    </div>
  </div>
</Card>
`;
